pkg/crds/crds.go

```diff
@@ -0,0 +1,192 @@
+	calicoOprtrCRDsRe = regexp.MustCompile(fmt.Sprintf("(%s)", strings.Join(calicoCRDNames, "|")))
+}
+
+func getCalicoCRDSource() map[string][]byte {
+	ret := map[string][]byte{}
+	entries, err := calicoCRDFiles.ReadDir("calico")
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read Calico CRDs: %v", err))
+	}
+
+	for _, entry := range entries {
+		b, err := calicoCRDFiles.ReadFile(path.Join("calico", entry.Name()))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to read Calico CRD %s: %v", entry.Name(), err))
+		}
+
+		if len(yamlDelimRe.FindAllString(string(b), -1)) > 1 {
+			panic(fmt.Sprintf("Too many yaml delimiters in Calico CRD %s", entry.Name()))
+		}
+
+		ret[entry.Name()] = yamlDelimRe.ReplaceAll(b, []byte("\n"))
+	}
+
+	return ret
+}
+
+func getEnterpriseCRDSource() map[string][]byte {
+	ret := map[string][]byte{}
+	entries, err := enterpriseCRDFiles.ReadDir("enterprise")
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read Enterprise CRDs: %v", err))
+	}
+
+	for _, entry := range entries {
+		b, err := enterpriseCRDFiles.ReadFile(path.Join("enterprise", entry.Name()))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to read Enterprise CRD %s: %v", entry.Name(), err))
+		}
+
+		if len(yamlDelimRe.FindAllString(string(b), -1)) > 1 {
+			panic(fmt.Sprintf("Too many yaml delimiters in Enterprise CRD %s", entry.Name()))
+		}
+
+		ret[entry.Name()] = yamlDelimRe.ReplaceAll(b, []byte("\n"))
+	}
+
+	return ret
+}
+
+func getOperatorCRDSource(variant opv1.ProductVariant) map[string][]byte {
+	ret := map[string][]byte{}
+	entries, err := operatorCRDFiles.ReadDir("operator")
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read Operator CRDs: %v", err))
+	}
+
+	for _, entry := range entries {
+		if variant == opv1.Calico {
+			if !calicoOprtrCRDsRe.MatchString(entry.Name()) {
+				continue
+			}
+		}
+
+		b, err := operatorCRDFiles.ReadFile(path.Join("operator", entry.Name()))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to read Operator CRD %s: %v", entry.Name(), err))
+		}
+
+		if len(yamlDelimRe.FindAllString(string(b), -1)) > 1 {
+			panic(fmt.Sprintf("Too many yaml delimiters in Operator CRD %s", entry.Name()))
+		}
+
+		ret[entry.Name()] = yamlDelimRe.ReplaceAll(b, []byte("\n"))
+	}
+
+	return ret
+}
+
+func GetCRDs(variant opv1.ProductVariant) []*apiextenv1.CustomResourceDefinition {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if variant == opv1.Calico {
+		if len(calicoCRDs) == 0 {
+			crdyamls := getCalicoCRDSource()
+			for _, yml := range crdyamls {
+
+				crd := &apiextenv1.CustomResourceDefinition{}
+				err := yaml.Unmarshal(yml, crd)
+				if err != nil {
+					fmt.Println(fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group))
+					fmt.Println(err)
+					return calicoCRDs
+				}
+				crd.Name = fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group)
+				calicoCRDs = append(calicoCRDs, crd)
+			}
+			crdyamls = getOperatorCRDSource(variant)
```

Comment:
This seems to be a somewhat duplicate block of code that might be worth pulling out into a separate function.